src/main/scala/sigmastate/serialization/ValueSerializer.scala

```diff
@@ -75,12 +32,12 @@ object ValueSerializer
     RelationSerializer(EqCode, EQ.applyNonTyped, Seq(Constraints.sameType2)),
     RelationSerializer(NeqCode, NEQ.apply, Seq(Constraints.sameType2)),
 
-    TwoArgumentsSerializerChecked(XorCode, Xor.apply),
-    TwoArgumentsSerializerChecked(AppendBytesCode, AppendBytes.apply),
-    TwoArgumentsSerializer(ExponentiateCode, Exponentiate.apply, Seq(Constraints.sameType2)),
-    TwoArgumentsSerializerChecked(MultiplyGroupCode, MultiplyGroup.apply),
-    TwoArgumentsSerializerChecked(MinusCode, Minus.apply),
-    TwoArgumentsSerializerChecked(PlusCode, Plus.apply),
+    TwoArgumentsSerializer(XorCode, Xor.apply),
+    TwoArgumentsSerializer(AppendBytesCode, AppendBytes.apply),
+    TwoConstrainedArgumentsSerializer(ExponentiateCode, Exponentiate.apply, Seq(Constraints.sameType2)),
```

Comment:
Why cant' TwoArgumentsSerializer and TwoConstrainedArgumentsSerializer be combined via optional constructor argument?